src/graph/LookupExecutor.cpp

```diff
@@ -0,0 +1,499 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "graph/LookupExecutor.h"
+
+namespace nebula {
+namespace graph {
+
+LookupExecutor::LookupExecutor(Sentence *sentence, ExecutionContext *ectx)
+    : TraverseExecutor(ectx, "lookup") {
+    sentence_ = static_cast<LookupSentence*>(sentence);
+}
+
+Status LookupExecutor::prepare() {
+    return Status::OK();
+}
+
+Status LookupExecutor::prepareClauses() {
+    DCHECK(sentence_ != nullptr);
+    Status status = Status::OK();
+    spaceId_ = ectx()->rctx()->session()->space();
+    expCtx_ = std::make_unique<ExpressionContext>();
+    expCtx_->setStorageClient(ectx()->getStorageClient());
+    expCtx_->setSpace(spaceId_);
+    do {
+        status = checkIfGraphSpaceChosen();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareFrom();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareWhere();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareYield();
+        if (!status.ok()) {
+            break;
+        }
+    } while (false);
+
+    if (!status.ok()) {
+        LOG(ERROR) << "Preparing failed: " << status;
+        return status;
+    }
+    return status;
+}
+
+Status LookupExecutor::prepareFrom() {
+    from_ = sentence_->from();
+    auto *mc = ectx()->getMetaClient();
+    auto tagResult = mc->getTagIDByNameFromCache(spaceId_, *from_);
```

Comment:
Do we have forbidden same name in tag and edge?

Reply:
> Do we have forbidden same name in tag and edge?

Yes, the tag name and edge names are unique. so  it's ok.